reactor-netty-http/src/main/java/reactor/netty/http/server/DefaultHttpServerRoutes.java

```diff
@@ -104,7 +107,30 @@ public HttpServerRoutes route(Predicate<? super HttpServerRequest> condition,
 			return Mono.error(t); //500
 		}
 
-		return response.sendNotFound();
+		if (matchHandlers.isEmpty()) {
+			return response.sendNotFound();
+		}
+
+		// only one handler match request,just apply for speed
+		if (matchHandlers.size() == 1) {
+			return matchHandlers.get(0).apply(request, response);
+		}
+
+		for (HttpRouteHandler handler : matchHandlers) {
+			// if condition is not HttpPredicate,have no method to compare
+			if (!(handler.condition instanceof HttpPredicate)) {
+				return matchHandlers.get(0).apply(request, response);
+			}
+		}
+
+		matchHandlers.sort((o1, o2) -> {
```

Comment:
I wonder if it would make sense to offer customization here.
A) an option for the user to set the `Comparator` to use
B) an option for the user to opt out of sorting (eg. by providing `null` as the `Comparator`)

Reply:
Also maybe we could make the `HttpRouteHandler` itself `Comparable`?
it could delegate to `HttpPredicate` in case both sides have such a predicate, or globally prioritize barebone `Predicate`s over `HttpPredicate`s ?

@violetagg in which case can we have a simple `Predicate` as the condition, btw?

Reply:
> I wonder if it would make sense to offer customization here.
> A) an option for the user to set the `Comparator` to use
> B) an option for the user to opt out of sorting (eg. by providing `null` as the `Comparator`)

I agree with you.

Reply:
@ctlove0523 Can we introduce a configuration so that by default we keep the current behaviour and then if someone wants to configure ordering then this feature to be enabled

Reply:
@violetagg ok i will try